consensus/hotstuff/pacemaker/timeout/controller.go

```diff
@@ -74,22 +68,22 @@ func (t *Controller) Channel() <-chan time.Time {
 }
 
 // StartTimeout starts the timeout of the specified type and returns the timer info
-func (t *Controller) StartTimeout(ctx context.Context, view uint64) *model.TimerInfo {
+func (t *Controller) StartTimeout(ctx context.Context, view uint64) model.TimerInfo {
 	t.stopTicker() // stop old timeout
 
 	// setup new timer:
 	// when round duration is small react with faster timeout rebroadcasts
 	duration := t.ReplicaTimeout()
-	tickInterval := time.Duration(math.Min(float64(duration.Milliseconds()), t.cfg.MaxTimeoutObjectRebroadcastInterval))
-	t.timerInfo = &model.TimerInfo{View: view, StartTime: time.Now().UTC(), Duration: duration}
+	tickInterval := time.Duration(math.Min(float64(duration.Milliseconds()), t.cfg.MaxTimeoutObjectRebroadcastInterval) * float64(time.Millisecond))
```

Comment:
To me, calculating the timeout values seems quite clunky due to the repeated conversations 
* from `time.Duration` -> `float64` for the purpose of numerical computations
* `float64` -> Duration for the purpose of _internal_ passing of values

The fact that we forgot the scaling factor `float64(time.Millisecond)` here is an indicator to me that the API is too error prone. 

### Suggestion:
[`ReplicaTimeout()`](https://github.com/onflow/flow-go/blob/e96e75f2aca1837728a2305104e35ed2064268d1/consensus/hotstuff/pacemaker/timeout/controller.go#L122) is only used internally within the `timeout.Controller`.  Therefore, I feel we should tune the return type to improve clarity. My suggestion would be:
*  rename `ReplicaTimeout()` to `replicaTimeout` and return the timeout as `float64` in (units of Milliseconds)
    ```golang
    // ReplicaTimeout returns the duration [in MILLISECONDS] of the current view before we time out in.
    func (t *Controller) replicaTimeout() float64 {
    	if t.r <= t.cfg.HappyPathMaxRoundFailures {
    		return t.cfg.MinReplicaTimeout
    	}
    	r := float64(t.r - t.cfg.HappyPathMaxRoundFailures)
    	if r >= t.maxExponent {
    		return t.cfg.MaxReplicaTimeout
    	}
    	// compute timeout duration [in milliseconds]:
    	return t.cfg.MinReplicaTimeout * math.Pow(t.cfg.TimeoutAdjustmentFactor, r)
    }
    ```
* only after all timings are computed as `float64` (by convention in milliseconds), I would convert to `time.Duration`:
    ```golang
    // StartTimeout starts the timeout of the specified type and returns the timer info
    func (t *Controller) StartTimeout(ctx context.Context, view uint64) model.TimerInfo {
    	t.stopTicker() // stop old timeout

    	// setup new timer
    	durationMs := t.replicaTimeout()                                                         // duration of current view in units of Milliseconds
    	rebroadcastIntervalMs := math.Min(durationMs, t.cfg.MaxTimeoutObjectRebroadcastInterval) // time between attempted re-broadcast of timeouts if there is no progress
    	t.timeoutChannel = make(chan time.Time, 1)                                               // channel for delivering timeouts

    	// start timeout logic for (re-)broadcasting timeout objects on regular basis as long as we are in the same round.
    	var childContext context.Context
    	childContext, t.stopTicker = context.WithCancel(ctx)
    	duration := time.Duration(durationMs) * time.Millisecond
    	rebroadcastInterval := time.Duration(rebroadcastIntervalMs) * time.Millisecond
    	go tickAfterTimeout(childContext, duration, rebroadcastInterval, t.timeoutChannel)

    	return model.TimerInfo{View: view, StartTime: time.Now().UTC(), Duration: duration}
    }
    ```